### DISCUSSION TITLE
Issue with .htaccess and .html extensions in SSG site

### DISCUSSION THREAD

**Discussion by Gusterson** [vercel/next.js#14133] [Help]
I'm exporting my site to be fully static. This generates a bunch of .html files. Cool.

I'm using an .htaccess file to hide the .html extensions:

```
RewriteEngine on
RewriteCond %{REQUEST_FILENAME} !-d
RewriteCond %{REQUEST_FILENAME}\.html -f
RewriteRule ^(.*)$ $1.html
```
Everything works flawlessly, EXCEPT:

When I navigate to the "index" page in a folder with pages created via a dynamic routing outside of my app. Example:

In my next app, I have folder:

```
/articles/
   [slug].js
   index.js
```
In various pages, I use Link:
```
import Link from 'next/link';

<Link href="/articles" as="/articles">
            <a>Articles</a>
</Link>
```
This takes me to `localhost/articles` and displays my `index.js` generated page. However, if I refresh, or use the back-button (to return back to the page from another one) all I get is a list of the generated `.html` files in the directory.

I now see that the `articles.html` file generated by `index.js` lives in the root folder, and there is not `index.html` living in the `/articles/` folder... this seems odd to me.

So...

- How do I fix this?
- Is this an .htaccess issue? A nextjs config issue?
- Wouldn't it be better for NextJS to create a `articles\index.html` instead of a file in the root directory? 

I tried playing around with `exportTrailingSlash` which seems related, but this created other problems like always having a trailing slash at the end of all my links ;-P

Anyway, if someone can point me to a straightforward example of how to make SSG work as expected, that would awesome. Thanks!

**Comment by vinocrzy** [comment]
**Use this `.htaccess` file to fix this problem** 

``` apacheconf
RewriteEngine On
RewriteRule ^([^/]+)/$ $1.html
RewriteCond %{REQUEST_FILENAME} !-f
RewriteRule ^([^/]+)/$ $1.html
RewriteRule ^([^/]+)/([^/]+)/$ /$1/$2.html
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
RewriteCond %{REQUEST_URI} !(\.[a-zA-Z0-9]{1,5}|/)$
RewriteRule (.*)$ /$1/ [R=301,L]
```

**Reply by Pherwerz** [reply]
it doesn't solve the problem of dynamic route

**Comment by ajaykumar010** [comment]
its Does not work proparly With Daynamic Routing With Slug

**Comment by 0zuna** [comment]
@Gusterson did you find a solution?
